gym-roboy/envs/ros_proxy.py

```diff
@@ -60,42 +60,67 @@ def __init__(self):
         self.node = rclpy.create_node('gym_client')
         self.step_cli = self.node.create_client(GymStep, 'gym_step')
         self.reset_cli = self.node.create_client(GymReset, 'gym_reset')
+        self.step_size = 1.0;
     
     def forward_reset_command(self):
-        step_size = 1.0
-        req = GymStep.Request()
-        res = GymStep.Response()
-        req.step_size = step_size
+
+        while not self.reset_cli.wait_for_service(timeout_sec=1.0):
+            self.node.get_logger().info('service not available, waiting...')
+
+        req = GymReset.Request()
+
         future = self.reset_cli.call_async(req)
         rclpy.spin_until_future_complete(self.node,future)
         if future.result() is not None:
-            self.node.get_logger().info("result: %f" % future.result().q[1])
-        q = future.result()
-        return MsjRobotState(joint_angle=q.q, joint_vel=q.qdot)
+            q = future.result().q
+            qd = future.result().qdot
+            q_str = str(q).strip('[]')
+            qd_str = str(qd).strip('[]')
+            self.node.get_logger().info("Reset, joint angles: %s" % q_str)
+            self.node.get_logger().info("Reset, joint velocity: %s" % qd_str)
+
+        robot_state = future.result()
+        return MsjRobotState(joint_angle=robot_state.q, joint_vel=robot_state.qdot)
 
     def forward_step_command(self, action):
 
         while not self.step_cli.wait_for_service(timeout_sec=1.0):
             self.node.get_logger().info('service not available, waiting...')
 
-        step_size = 1.0
+
         req = GymStep.Request()
         res = GymStep.Response()
         req.set_points = action
-        req.step_size = step_size
+        req.step_size = self.step_size
         future = self.step_cli.call_async(req)
         rclpy.spin_until_future_complete(self.node,future)
         if future.result() is not None:
-            self.node.get_logger().info("result: %f" % future.result().q[1])
-        q = future.result()
-        return MsjRobotState(joint_angle=q.q, joint_vel=q.qdot)
+            q = future.result().q
```

Comment:
Would it be a good idea to rename all `q`s and `q_dot` to `joint_position` and `joint_vel` ?
At least for me, its easier to understand it. Also, `Q` is already used a lot in RL in the context of https://en.wikipedia.org/wiki/Q-learning

Reply:
Up to you

Reply:
IMHO q and qdot is convention in robotics and it is used like that everywhere for example in CARDSflow code. However I see the point I checked openAI, how they used it.  They use qpos and qvel https://github.com/openai/gym/blob/master/gym/envs/robotics/fetch_env.py. But I am not sure how much of a trouble it can cause. Correct me if I'm wrong Qlearning uses capital Q, right?  Capital Q is never used to refer joint angles.

Reply:
@BarisYazici Makes sense